guides/common/assembly_configuring-keycloak-external-authentication-with-totp.adoc

```diff
@@ -1,9 +1,10 @@
 ifdef::context[:parent-context: {context}]
-:context: keycloak-totp
 
 [id="configuring-keycloak-authentication-with-totp_{context}"]
 = Configuring {Keycloak} Authentication with TOTP
 
+:context: keycloak-totp
```

Comment:
Should we not define the context above where the context will be invoked? To me this makes no sense to define it after it is called in the title. I think this is problematic?

Reply:
Yeah this is one of the quirks of nesting assemblies (for simplicity I'm referring to the master.adoc file as the top assembly).

The key here is that the context defined in the assembly is only used for the modules included in that assembly, just as the context in the master.adoc file is used for the assemblies included in that master.adoc file. 

So, this assembly should build its ID from the context of the assembly it's included in (master.adoc), and then define its own context for its included modules to build their IDs with. So, using the :parent-context: ifdefs to preserve and then reinstate the top assembly's context, it looks something like this (at least for visualisation purposes): 

```
Master Assembly (master.adoc)
:context: master

  include::assembly1.adoc
    ID=assembly1_master
    Assembly 1
    :context: assembly1
  
      include::module1.adoc
        ID=module1_assembly1
        Module 1
    
      include::module2.adoc
        ID=module2_assembly1
        Module 2
    
  include::assembly2.adoc
    ID=assembly2_master
    Assembly 2
    :context: assembly2
    
      include::module1.adoc
        ID=module1_assembly2
        Module 1
        
      include::module3.adoc
        ID=module3_assembly2
        Module 3
```
    
I hope that makes sense and isn't just gibberish! It's described (probably better) here: https://redhat-documentation.github.io/modular-docs/#nesting-assemblies

Reply:
Thank you for explaining this to me!